Sources/Rendering/Core/VolumeProperty/index.js

```diff
@@ -203,6 +203,30 @@ function vtkVolumeProperty(publicAPI, model) {
   publicAPI.getInterpolationTypeAsString = () =>
     macro.enumToString(InterpolationType, model.interpolationType);
 
+  publicAPI.getColorMixCode = () => {
+    if (model.customColorMixCode) {
+      return model.customColorMixCode;
+    }
+    switch (model.colorMixPreset) {
+      case ColorMixPreset.ADDITING:
+        return `
+          float opacity0 = goFactor.x * pwfValue0;
+          float opacity1 = goFactor.y * pwfValue1;
+          float opacitySum = opacity0 + opacity1;
+          if (opacitySum == 0.0) {
+            return vec4(0.0);
+          }
+          #if vtkLightComplexity > 0
+            applyLighting(tColor0, normal0);
+            applyLighting(tColor1, normal1);
+          #endif
+          vec3 mixedColor = mix(tColor0, tColor1, opacity1 / opacitySum);
```

Comment:
Maybe I'm not understanding the big picture here.

How is this different from the current code where independent components are alpha-blended?
Secondly, this is not doing an additive blend, is it?

Reply:
If we have alphas `a0` and `a1`, and colors `c0` and `c1`, then:
With alpha blending:
```
color = (c0 * a0 + (1 - a0) * c1 * a1 ) / alpha
alpha = a0 + (1 - a0) * a1
```
With additive mode:
```
color = (c0 * a0 + c1 * a1) / alpha
alpha = a0 + a1
```

Also how to create alpha blending with independent components in the current shader?

Do you suggest a different name instead or additive mode ?